### PR TITLE
🧑‍💻 Transition Unguard to Helm

### PR DESCRIPTION
**Pull request by MfCrizz**
🚀 This PR transitions **Unguard** to a **Helm Chart**. 🚀
Please have a look and try it yourself. Feedback is very appreciated! :smile: 

To get started please follow  `/chart/README.md`

Note: The individual commits will be squashed before merging.

### CODE REVIEW

File: docs/DEV-GUIDE.md [hunk 1]
```diff
@@ -23,6 +23,11 @@ This document explains how to build and run Unguard locally inside Kubernetes us
         kind create cluster --name unguard --config ./k8s-manifests/localdev/kind/cluster-config.yaml
         ```
 
+      * Optional add NGINX-Ingress
```
Comment (W3D3) [inline docs/DEV-GUIDE.md:26]: [ISSUE] Be more specific here
```suggestion
      * Optionally add an nginx ingress controller if you want to expose Unguard locally without port-forwarding
```

File: docs/DEV-GUIDE.md [hunk 2]
```diff
@@ -49,18 +54,11 @@ This document explains how to build and run Unguard locally inside Kubernetes us
 1. Use `skaffold` to build and deploy the application.
    The first full build will take up to 20 minutes.
    If the images should be rebuilt automatically, run `skaffold dev`.
+    > **Note**: If you deploy with **minikube** don't forget to forward the Docker daemon first (see above)
```
Comment (W3D3) [inline docs/DEV-GUIDE.md:57]: skaffold pretty reliably detects a running minikube and should adapt automatically, so not sure if we should repeat this here. Maybe even put the whole paragraph above into a troubleshooting section at the bottom.
Seee https://skaffold.dev/docs/environment/local-cluster/#auto-detection

File: docs/DEV-GUIDE.md [hunk 3]
```diff
@@ -81,18 +79,18 @@ This document explains how to build and run Unguard locally inside Kubernetes us
...
+skaffold delete -p localdev
 ```
 
 ## 🙋‍♀️ FAQ
 
 ### What are skaffold profiles?
 
 Have a look at the profiles that are supported in `skaffold.yaml`.
-Often, you might also want to also deploy Jaeger.
+Often, you might also want to also deploy **Jaeger**. [See these instructions](TRACING.md)
```
Comment (W3D3) [inline docs/DEV-GUIDE.md:90]: [CLARIFY] This looks misleading, the `tracing` profile would not install Jaeger? Maybe rephrase this?
```suggestion
To enable distributed tracing, install a tracing collector like **Jaeger** and use the `tracing` profile. [See these instructions.](TRACING.md)
```
Also, shouldn't we link here to docs/JAEGER-TRACING-DEPLOYMENT.md?

File: skaffold.yaml [hunk 4]
```diff
@@ -19,14 +19,14 @@ metadata:
 build:
   artifacts:
     # context is the directory with the Dockerfile for that image
+    - image: unguard-proxy-service
```
Comment (W3D3) [inline skaffold.yaml:22]: [CLARIFY] Why the reordering?

**Review by W3D3 — CHANGES_REQUESTED** [review]

File: skaffold.yaml (hunk 4 above)
Comment (MfCrizz) [reply skaffold.yaml:22]: It happened during the implementation but there is no reason.
Will be undone.

File: chart/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+package manager.
+
+## Prerequisites
+
+- Helm 3.8.0+
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+> **Note**: This chart presumes an already running and configured MariaDB database in the cluster
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply chart/README.md:15]: Good point!

File: docs/DEV-GUIDE.md (hunk 3 above)
Comment (MfCrizz) [reply docs/DEV-GUIDE.md:90]: Yes. TRACING.md also needs to be deleted/reworked

File: README.md [hunk 6]
```diff
@@ -43,11 +43,11 @@ Unguard is composed of eight microservices written in different languages that t
 
 ## 🖥️ Local Deployment
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply README.md]: Great idea :D

File: chart/values.yaml [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+      ports:
+        containerPort: 9091
+      env:
+        SERVER_PORT: 9091
+        MARIADB_SERVICE: unguard-mariadb
+        MARIADB_PASSWORD:
+          secretKeyRef:
+            name: unguard-mariadb
+            key: mariadb-root-password
+        JAEGER_AGENT_HOST: agent # change depending on your jaeger deployment
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply chart/values.yaml:176]: The name is derived from the name of the mariaDB installation. Yes, this would be a good idea.

**Review by W3D3 — CHANGES_REQUESTED** [review]

File: chart/templates/_helpers.tpl [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+{{- define "unguard.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply chart/templates/_helpers.tpl:22]: Yes, I also removed most of the auto generated methods.

File: .github/workflows/release.yaml [hunk 9]
```diff
@@ -53,3 +62,29 @@ jobs:
           tag: ${{ env.IMAGE_TAG }}
           repository: ${{ vars.REGISTRY }}/${{ github.repository }}
           push: true
+
+      - name: Update helm image tags
+        uses: jacobtomlinson/gha-find-replace@v3
+        with:
+          find: "tag: (.)*"
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply .github/workflows/release.yaml:69]: We need to change the default value of the image tags to their corresponding values for this build. This can be only done in the values.yaml. 
But we can replace the regex with a more safer variant " \btag:.*" -> This matches only if the value starts with "tag:...."
The other two replace action should be safe as there are no other values with the same name allowed. But here we can also replace the regex with a safer version.

File: skaffold.yaml [hunk 10]
```diff
@@ -41,18 +41,14 @@ build:
       context: src/status-service
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply skaffold.yaml:41]: As discussed, can be don in a later step :)

File: chart/README.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+## Parameters
+
+### Global parameters
+
+| Name                             | Description                                     | Default Value     |
+|----------------------------------|-------------------------------------------------|-------------------|
+| `localDev.enabled`               | Enable for local (minikube/kind) deployment     | `true`            |
+| `aws.enabled`                    | Enable for AWS cluster deployment               | `false`           |
```
[the remark this answers is not in the document]
Comment (MfCrizz) [reply chart/README.md]: Section is now reworked

File: .github/workflows/release.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (MfCrizz) [reply .github/workflows/release.yaml]: Is now replaced with `yaml-update-action` . But sadly a wildcard wasn't working, so now all image tags are explicitly listed.

**Review by W3D3 — APPROVED** [review]
LGTM! :smile:

**Comment by MfCrizz** [comment]
Hi Sarah, 
which version of Helm do you have installed?
Could you please try to install the chart adding the `--debug` flag?
`helm install unguard  oci://ghcr.io/dynatrace-oss/unguard/chart/unguard --wait --namespace unguard --create-namespace --debug`
Does the installation using the local chart work?
`helm install unguard ./chart --wait --namespace unguard --create-namespace`
